Samples/HelloWorld/OrleansClient/Program.cs

```diff
@@ -16,8 +16,27 @@ public class Program
         static int Main(string[] args)
         {
             var config = ClientConfiguration.LocalhostSilo();
-            const int initializeAttemptsBeforeFailing = 5;
-            int attemp = 0;
+            try
+            {
+                InitializeWithRetries(config, initializeAttemptsBeforeFailing: 5);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Orleans client initialization failed failed due to {ex}");
+
+                Console.ReadLine();
+                return 1;
+            }
+
+            DoClientWork().Wait();
+            Console.WriteLine("Press Enter to terminate...");
+            Console.ReadLine();
+            return 0;
+        }
+
+        private static void InitializeWithRetries(ClientConfiguration config, int initializeAttemptsBeforeFailing)
+        {
+            int attempt = 0;
             while (true)
             {
```

Comment:
we are catching a general 'Exception' in main, what's wrong at catching a general 'Exception' here. That way we make the code more precise. 
the exception handling in main can be moved here. and we should avoid wild loop while I understand this is something we inherit from the original sample.